userspace/libscap/scap_procs.c

```diff
@@ -1778,3 +1778,107 @@ int32_t scap_procfs_get_threadlist(struct scap_engine_handle engine, struct ppm_
 	return SCAP_FAILURE;
 #endif /* __linux__ */
 }
+
+int32_t scap_fd_scan_vtable(scap_t *handle, const scap_threadinfo *src_tinfo, scap_threadinfo *dst_tinfo, char *error)
+{
+	uint64_t n_fdinfos, i;
+	const scap_fdinfo *fdinfos;
+	scap_fdinfo *fdi = NULL;
+	uint32_t res;
+
+	res = handle->m_vtable->get_fdinfos(handle->m_engine, src_tinfo, &n_fdinfos, &fdinfos);
+	if (res != SCAP_SUCCESS)
+	{
+		return res;
+	}
+
+	for (i = 0; i < n_fdinfos; i++)
+	{
+		res = scap_fd_allocate_fdinfo(handle, &fdi, fdinfos[i].fd, fdinfos[i].type);
+		if (res != SCAP_SUCCESS)
+		{
+			snprintf(error, SCAP_LASTERR_SIZE, "can't allocate scap fd handle for file fd %" PRIu64, fdinfos[i].fd);
+			return res;
+		}
+
+		// copy the contents
+		*fdi = fdinfos[i];
+
+		res = scap_add_fd_to_proc_table(handle, dst_tinfo, fdi, error);
+		if (res != SCAP_SUCCESS)
+		{
+			scap_fd_free_fdinfo(&fdi);
+			continue;
+		}
+
+		if(handle->m_proc_callback != NULL)
```

Comment:
I don't understand this part; care to clarify? 
Thanks!

Reply:
This is due to how `libsinsp` and `libscap` currently interact when reading proc and fd tables.

Essentially, if libsinsp registered a callback with libscap the fd or proc value will be passed straight to libsinsp and discarded (free'd) which is what is happening here. Otherwise libscap will add it to its own table and keep it.

Reply:
Uh thanks, couldn't recall this! Thanks :)